core/src/main/java/uk/gov/gchq/koryphe/impl/predicate/AreIn.java

```diff
@@ -80,6 +82,10 @@ public void setAllowedValues(final Object[] allowedValuesArray) {
 
     @Override
     public boolean test(final Collection<?> input) {
+        if (isNullOrEmptyAllowed) {
+            return (!(null == allowedValues || allowedValues.isEmpty()));
```

Comment:
Instead of a separate return statement, this logic could be added into the one below. An example for the null bit:
`return (allowedValues == null && isNullOrEmptyAllowed) || ...`
These may need to be put onto newlines as it will get quite long

Reply:
I added a separate return Statement because writing 

(return (allowedValues == null && isNullOrEmptyAllowed) || ...)

was allowing empty input to return true even if flag was false.